sdk/metric/controller/basic/config.go

```diff
@@ -66,7 +66,8 @@ type Option interface {
 
 // WithResource sets the Resource configuration option of a Config.
 func WithResource(r *resource.Resource) Option {
-	return resourceOption{r}
+	res := resource.Merge(resource.Environment(), r)
```

Comment:
```suggestion
	res := resource.Merge(r, resource.Environment())
```

I think we want the attributes from the environment to take precedence.

Reply:
According to the specification:

> The SDK MUST extract information from the OTEL_RESOURCE_ATTRIBUTES environment variable and merge this, as the secondary resource, with any resource information provided by the user, i.e. the user provided resource information has higher priority.

So I used the Merge function in this manner